src/providers/local/join-pool.provider.ts

```diff
@@ -264,13 +289,16 @@ const provider = (props: Props) => {
 
     try {
       joinPoolService.setJoinHandler(isSingleAssetJoin.value);
+      const simulationType = getSimulationType();
 
       const output = await joinPoolService.queryJoin({
         amountsIn: amountsInWithValue.value,
         tokensIn: tokensIn.value,
         prices: prices.value,
         signer: getSigner(),
         slippageBsp: slippageBsp.value,
+        relayerSignature: relayerSignature.value,
+        simulationType: simulationType,
```

Comment:
At a high level, I don't think we should be specifying the simulationType and passing it into the service from the provider. It's something specific to a particular handler and should be handled there. I think it makes more sense that approval state becomes part of the JoinParams and then the handler decides which simulation type to use.

Reply:
There may also be other handlers in the future that need access to the approval state to decide how to handle the join but it wouldn't require simulation type so then you'd have to pass both anyway.

Reply:
Done. Provider decides the simulation type based on token approvals, and account's token balances.